src/modules/auth/components/ChooseAvatar/chooseAvatar.js

```diff
@@ -1,103 +1,113 @@
-import React from 'react';
-import { withTranslation } from 'react-i18next';
+import React, { useState, useEffect, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
 import { Link } from 'react-router-dom';
 import routes from 'src/routes/routes';
-import { PrimaryButton, TertiaryButton } from 'src/theme/buttons';
+import { PrimaryButton, TertiaryButton } from '@theme/buttons';
+import useSettings from '@settings/hooks/useSettings';
+import { extractKeyPair, extractAddressFromPublicKey } from '@wallet/utils/account';
+import { defaultDerivationPath } from '@account/const';
 import WalletVisual from '@wallet/components/walletVisual';
 import registerStyles from '../Signup/register.css';
 import styles from './chooseAvatar.css';
 
-class ChooseAvatar extends React.Component {
-  constructor() {
-    super();
+// eslint-disable-next-line max-statements
+const ChooseAvatar = ({ accounts, selected, handleSelectAvatar, nextStep }) => {
+  const [deselect, setDeselect] = useState({});
+  const [accountList, setAccountList] = useState([]);
+  const prevDeselect = useRef({});
+  const wrapperRef = useRef();
+  const timeout = useRef();
+  const { t } = useTranslation();
+  const { enableAccessToLegacyAccounts } = useSettings('enableAccessToLegacyAccounts');
 
-    this.state = {
-      deselect: {},
-    };
+  const getAvatarAnimationClassName = ({ address, selected: selectedAddress, previous }) =>
+    selectedAddress === address ? styles.selected : (previous === address && styles.deselect) || '';
 
-    this.getAvatarAnimationClassName = this.getAvatarAnimationClassName.bind(this);
-    this.handleNextStep = this.handleNextStep.bind(this);
-    this.setWrapperRef = this.setWrapperRef.bind(this);
-  }
+  useEffect(() => {
+    if (accounts.length) {
+      const getAcctList = async () => {
+        await Promise.all(
+          accounts.map(async (account) => {
+            const options = {
+              passphrase: account.passphrase,
+              enableAccessToLegacyAccounts,
```

Comment:
remove this, it will be false by default.

Reply:
@ManuGowda Done